azure-iot-device/azure/iot/device/iothub/pipeline/mqtt_topic_iothub.py

```diff
@@ -257,7 +257,8 @@ def encode_properties(message_to_send, topic):
     topic += system_properties_encoded
 
     if message_to_send.custom_properties and len(message_to_send.custom_properties) > 0:
-        topic += "&"
+        if system_properties and len(system_properties) > 0:
```

Comment:
>if system_properties and len(system_properties) > 0: [](start = 8, length = 52)

There should be a unit test to validate this

Reply:
And,  for what it's worth, I've always thought that mqtt_topic_iothub.py deserves to have its own set of unit tests.

---
In reply to: [308970175](https://github.com/Azure/azure-iot-sdk-python-preview/pull/153#discussion_r308970175) [](ancestors = 308970175)

Reply:
the tests are now params of the iothub mqtt stages file..a separate test file does not exist 

---
In reply to: [308971540](https://github.com/Azure/azure-iot-sdk-python-preview/pull/153#discussion_r308971540) [](ancestors = 308971540,308970175)

Reply:
At some point we should probably start testing them separately though. I don't think this PR is the right place for it, but it would be good to do. I'll add a task to the taskboard.

---
In reply to: [309413760](https://github.com/Azure/azure-iot-sdk-python-preview/pull/153#discussion_r309413760) [](ancestors = 309413760,308971540,308970175)